File: app/src/main/java/com/example/neon/deardiary/settings/SettingsPresenter.java
package com.example.neon.deardiary.settings;

import com.example.neon.deardiary.data.source.DataSource;

class SettingsPresenter implements SettingsContract.Presenter {
    private DataSource mDataSource;
    private SettingsContract.View mView;

    SettingsPresenter(DataSource dataSource, SettingsContract.View view) {
        mDataSource = dataSource;
        mView = view;
        mView.setPresenter(this);
    }

    @Override
    public void start() {

    }

    @Override
    public void deleteAll() {
        mDataSource.deleteAll();
        mView.onDiaryDeleted();
    }
}

File: app/src/main/java/com/example/neon/deardiary/settings/SettingsPresenter.kt
package com.example.neon.deardiary.settings

import com.example.neon.deardiary.data.DataSource

internal class SettingsPresenter(private val mDataSource: DataSource?, private val mView: SettingsContract.View) : SettingsContract.Presenter {

    init {
        mView.setPresenter(this)
    }

    override fun deleteAll() {
        mDataSource !!.deleteAll()
        mView.onDiaryDeleted()
    }
}
